### ISSUE TITLE
[Feature Request] Expr - Support bitwise operators

### ISSUE THREAD

**Issue by adworacz** [vapoursynth/vapoursynth#904]
After getting confused for a few minutes, I discovered that Expr supports logical "and" but not a bitwise and, "&".

I encountered this while looking at updating LUTDerainbow with 16-bit support. It currently uses LUT2 to perform its bitwise and/&, as can be seen here: https://github.com/HomeOfVapourSynthEvolution/havsfunc/blob/master/havsfunc.py#L3453-L3457

Because its using LUT2, it's limited to handling clips no larger than 10 bits.

My thought was to work around this by using Expr instead of LUT2 for higher bit depths. Except that Expr doesn't support bitwise operators like and/or/xor, etc.

**Comment by sekrit-twc** [comment]
Expr operates on floating-point numbers. Bitwise operators don't mean anything in floating-point.

**Comment by adworacz** [comment]
Ahhh, that's something I didn't realize.

Dang.

So there's really a hard limit on the max bit depth supported for performing this kind of bitwise operation, right? LUT2 states that it can't run with more than a total of 20 bits, and Expr doesn't work with integers/bitwise logic.

Does that mean that something like Exprcpp is the only real solve here?

https://github.com/Endilll/exprcpp

**Comment by sekrit-twc** [comment]
Well, take a step back:

````
    umask = average_u.std.Binarize(threshold=21 << shift)
    vmask = average_v.std.Binarize(threshold=21 << shift)
    themask = core.std.Lut2(umask, vmask, function=lambda x, y: x & y)
    if y:
        umask = core.std.Lut2(umask, average_y, function=lambda x, y: x & y)
        vmask = core.std.Lut2(vmask, average_y, function=lambda x, y: x & y)
        themask = core.std.Lut2(themask, average_y, function=lambda x, y: x & y)

    output_u = core.std.MaskedMerge(input_u, fixed_u, themask if linkUV else umask)
````

`Lut2(umask, vmask, x & y)` is equivalent to "x + y", because umask and vmask are binarized. 

`core.std.Lut2(umask, average_y, x & y)` is equal to `MaskedMerge(BlankClip(), average_y, umask)`, again because umask is binarized.

etc.

**Comment by adworacz** [comment]
Ahhh that's a great point! Thanks for your help! 

I'll look into getting this merged upstream so that we can properly support 16-bit.

**Comment by Setsugennoao** [comment]
Just like https://github.com/vapoursynth/vapoursynth/issues/703, Akarin implemented this feature right away in v0.95, with https://github.com/AkarinVS/vapoursynth-plugin/commit/530b489435334199354c6e81b793327598a341f1

**Comment by myrsloik** [comment]
Outside the scope of expr, maybe I'll try some other ideas as separate plugins where there's a real compiler involved one day.
